content/post/52-go-ipfs-0.4.18.md

```diff
@@ -0,0 +1,283 @@
+[ipfs/go-bitswap#5](https://github.com/ipfs/go-bitswap/pull/5). While this won't
+help when transferring large files (with large blocks), this should help when
+transferring many tiny files.
+
+## Refactors and Endeavors
+
+This release saw yet another commands-library refactor, work towards the
+CoreAPI, and the first step towards reliable base32 CID support.
+
+### Commands Lib
+
+We've completely refactored our commands library (again). While it still needs
+quite a bit of work, it now requires significantly less boilerplate and should
+be significantly more robust. The refactor immediately found two broken tests
+and probably fixed quite a few bugs around properly returning and handling
+errors.
+
+### CoreAPI
+
+CoreAPI is a new way to interact with IPFS from Go. While it's still not
+final, most things you can do via the CLI or HTTP interfaces, can now be done
+through the new API.
+
+Currently there is only one implementation, backed by go-ipfs node, and there are
+plans to start http-api backed one soon. We are also looking into creating RPC
+interface using this API, which could help performance in some use cases.
+
+You can track progress in https://github.com/ipfs/go-ipfs/issues/4498
+
+### CIDv1/Base32 Migration
+
+Currently, IPFS is usually used in browsers by browsing to `https://SOME_GATEWAY/ipfs/CID/...`. There are two significant drawbacks to this approach:
+
+1. From a browser security standpoint, all IPFS "sites" will live under the same
+   origin (SOME_GATEWAY).
+2. From a UX standpoint, this doesn't feel very "native" (even if the gateway is
+   a local IPFS node).
+
+To fix the security issue, we intend to switch IPFS gateway links
+`https://ipfs.io/ipfs/CID` to to `https://CID.ipfs.dweb.link`. This way, the CID
+will be a part of the
+["origin"](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin) so
+each IPFS website will get a separate security origin.
+
+To fix the UX issue, we've been working on adding support for `ipfs://CID/...`
+to web browsers through our
+[ipfs-companion](https://github.com/ipfs/ipfs-companion/) add-on and some new,
+experimental extension APIs from Mozilla. This has the same effect of putting
+the CID in the URL origin but has the added benefit of looking "native".
+
+Unfortunately, origins must be *case insensitive*. Currently, most CIDs users
+see are *CIDv0* CIDs (those starting with `Qm`) which are *always* base58
+encoded and are therefore case-sensitive.
+
+Fortunately, CIDv1 (the latest CID format) supports arbitrary bases using the
+[multibase](https://github.com/multiformats/multibase/) standard. Unfortunately,
+IPFS has always treated equivalent CIDv0 and CIDv1 CIDs as distinct. This means
+that files added with CIDv0 CIDs (the default) can't be looked up using the
+equivalent CIDv1.
+
+This release makes some significant progress towards solving this issue by
+introducing two features:
+
+(1) The previous mentioned `ipfs cid base32` command for converting CID to a
```

Comment:
add "any" and bold it?
>> converting **any** CID to a case...

Reply:
Are we talking about CIDv0 in particular? If so we should just keep referencing that all along, and mention it's converting to CIDv1 in particular